mmocr/datasets/pipelines/transform_wrappers.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import inspect
+import random
+
+import mmcv
+import numpy as np
+import torchvision.transforms as transforms
+from mmcv.utils import build_from_cfg
+from mmdet.datasets.builder import PIPELINES
+from mmdet.datasets.pipelines import Compose
+from PIL import Image
+
+
+@PIPELINES.register_module()
+class OneOf:
+    """Randomly select and apply one of the transforms, each with the equal
+    chance.
+
+    Warning:
+        Different from albumentations, this wrapper only runs the selected
+        transform, but doesn't guarantee the transform can always be applied to
+        the input if the transform comes with a probability to run.
+
+    Args:
+        transforms (list[dict]): Candidate transforms to be applied.
+    """
+
+    def __init__(self, transforms):
+        assert isinstance(transforms, list) or isinstance(transforms, tuple)
+        assert len(transforms) > 0
+        self.transforms = []
+        for t in transforms:
+            if isinstance(t, dict):
+                self.transforms.append(build_from_cfg(t, PIPELINES))
+            elif callable(t):
+                self.transforms.append(t)
+            else:
+                raise TypeError('transform must be callable or a dict')
+
+    def __call__(self, results):
+        return random.choice(self.transforms)(results)
+
+    def __repr__(self):
+        repr_str = self.__class__.__name__
+        repr_str += f'(transforms={self.transforms})'
+        return repr_str
+
+
+@PIPELINES.register_module()
+class RunWithProb:
```

Comment:
TransWithSameProb. RunWithProb may have no same name style as other transfroms in MM

Reply:
RandomWrapper looks like a Random.Random()' wrapper, which can not know the function with name

Reply:
It should be fine as long as enough docstring is presented

Reply:
It may be not a good way when users look at this class in the config, they need to search the class name in the mmocr and read the docstring for understanding what is the class's function

Reply:
I agree that this is the ultimate goal, but we can stick to this if we both cannot come up a better name